File: python/app/common/Msg.py
from app.common.Base import Base
from app.common.Inc import Inc
import json

# 消息类
class Msg(Base) :

  __fds = {}
  __token = {}

  # 路由
  async def router(self,fds,socket,msg,token) :
    # 参数
    self.__fds = fds
    self.__token = token
    # 数据
    data = Inc.json_decode(msg)
    if not data :
      try : await socket.send(self.getJSON({'code':4000,'msg':'格式错误!'}))
      except Exception as e: pass
    # 消息
    elif data['type']=='msg' : await self.msg(socket,data)
    # 心跳
    else : await socket.send(self.getJSON({'type':'','code':0,'msg':'成功'}))

  # 消息
  async def msg(self,socket,data) :
    # 数据
    print(self.__token,data)
    # 指定用户
    if data['uid'] in self.__fds.keys() :
      server = self.__fds[data['uid']]
      await server.send(self.getJSON({'type':'msg','code':0,'msg':data['msg']}))

File: java/src/main/java/vip/webmis/java/common/Msg.java
package vip.webmis.java.common;

import java.io.IOException;
import java.util.HashMap;
import java.util.concurrent.ConcurrentHashMap;
import javax.websocket.Session;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;

/* 消息类 */
public class Msg extends Base {

  private static ConcurrentHashMap<String, Session> __fds = new ConcurrentHashMap<>();
  private static JSONObject __token = null;
  private static HashMap<String, Object> res = null;

  /* 路由 */
  public static void router(ConcurrentHashMap<String, Session> fds, Session socket, String msg, JSONObject token) throws IOException {
    // 参数
    __fds = fds;
    __token = token;
    // 数据
    JSONObject data = Inc.json_decode(msg);
    if (data == null || data.isEmpty()) {
      res = new HashMap<String, Object>();
      res.put("code", 4000);
      res.put("msg", "格式错误!");
      socket.getBasicRemote().sendText(JSON.toJSONString(res));
    }
    // 消息
    else if (data.get("type").equals("msg"))
      msg(socket, data);
    // 心跳
    else {
      res = new HashMap<String, Object>();
      res.put("type", "");
      res.put("code", 0);
      res.put("msg", "成功");
      socket.getBasicRemote().sendText(JSON.toJSONString(res));
    }
  }

  /* 消息 */
  public static void msg(Session socket, JSONObject data) throws IOException {
    // 数据
    System.out.println(__token);
    System.out.println(data);
    // 指定用户
    if(__fds.containsKey(data.get("uid"))){
      res = new HashMap<String,Object>();
      res.put("type","msg");
      res.put("code",0);
      res.put("msg","消息");
      __fds.get(data.get("uid")).getBasicRemote().sendText(JSON.toJSONString(res));
    }
  }
}
